hsm/asymmetric_decryption_key.go

```diff
@@ -0,0 +1,62 @@
+package hsm
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"github.com/ThalesIgnite/crypto11"
+	"github.com/ThalesIgnite/gose"
+	"github.com/ThalesIgnite/gose/jose"
+)
+
+// AsymmetricDecryptionKey implements RSA OAEP using SHA1 decryption.
+type AsymmetricDecryptionKey struct {
+	kid string
+	ctx *crypto11.Context
+	key crypto11.SignerDecrypter
+}
+
+// Kid the unique identifier of this key.
+func (a *AsymmetricDecryptionKey) Kid() string {
+	return a.kid
+}
+
+// Certificates associated x509 certificates.
+func (a *AsymmetricDecryptionKey) Certificates() []*x509.Certificate {
+	// TODO: lookup certificates
+	cert, err := a.ctx.FindCertificate([]byte(a.kid), nil, nil)
+	if err != nil {
+		// :thinking:
```

Comment:
We probably need to break the `CertifiableKey` interface to propagate the error.

Reply:
The introduction of a southbound API returning an error to .Certificates() can indeed break it.  As this is crypto11/HSM specific piece of code, could we have it be a panic(err), and mark this for a breaking change in the next major version?

Reply:
Done in f2d9de4334f85808a03d3e552fe5e801d9cc323d